docs/topics/coroutines-basics.md

```diff
@@ -1,293 +1,612 @@
+
+## Coroutine dispatchers
+
+A [coroutine dispatcher](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/#) controls which thread or thread pool coroutines use for their execution.
+Coroutines aren't always tied to a single thread.
+They can pause on one thread and resume on another, depending on the dispatcher.
+This lets you run many coroutines at the same time without allocating a separate thread for every coroutine.
+
+> Even though coroutines can suspend and resume on different threads,
+> values written before the coroutine suspends are still guaranteed to be available within the same coroutine when it resumes.
+>
+{style="tip"}
+
+A dispatcher works together with the [coroutine scope](#coroutine-scope-and-structured-concurrency) to define when coroutines run and where they run.
+While the coroutine scope controls the coroutine's lifecycle, the dispatcher controls which threads are used for execution.
+
+> You don't have to specify a dispatcher for every coroutine.
+> By default, coroutines inherit the dispatcher from their parent scope.
+> You can specify a dispatcher if you need to run a coroutine in a different context.
+> 
+> If the coroutine context doesn't include a dispatcher, coroutine builders use `Dispatchers.Default`.
 >
 {style="note"}
 
-Both pieces of code inside `launch { ... }` blocks execute _concurrently_, with 
-`World 1` printed first, after a second from start, and `World 2` printed next, after two seconds from start.
-A [coroutineScope][_coroutineScope] in `doWorld` completes only after both are complete, so `doWorld` returns and 
-allows `Done` string to be printed only after that:
+The `kotlinx.coroutines` library includes different dispatchers for different use cases.
+For example, [`Dispatchers.Default`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-default.html) runs coroutines on a shared pool of threads performing work in the background,
+separate from the main thread.
 
-```text
-Hello
-World 1
-World 2
-Done
-```
+To specify a dispatcher for a coroutine builder like `CoroutineScope.launch()`, pass it as an argument:
 
-<!--- TEST -->
+```kotlin
+suspend fun runWithDispatcher() = coroutineScope { // this: CoroutineScope
+    this.launch(Dispatchers.Default) {
+        println("Running on ${Thread.currentThread().name}")
+    }
+}
+```
 
-## An explicit job
+Alternatively, you can use a `withContext()` block to run all code in it on a specified dispatcher.
 
-A [launch] coroutine builder returns a [Job] object that is a handle to the launched coroutine and can be 
-used to wait for its completion explicitly.
-For example, you can wait for the completion of the child coroutine and then print the "Done" string:
+The following example runs on `Dispatchers.Default`, which is optimized for CPU-intensive operations like data processing:
```

Comment:
This section doesn't read fluently to me. "`Default` is a dispatcher based on a thread pool. Here's how you specify a dispatcher. `Default` is a dispatcher optimized for CPU-intensive operations." Either there are two explanations of what `Default` is, or there's one explanation interrupted by an example of a `launch` with a context.